### ISSUE TITLE
Is Policy.Priority ignored in the Query() command?

### ISSUE THREAD

**Issue by wellle** [aerospike/aerospike-client-go#178]
We're currently running some tests to evaluate using the `Query()` command. Our code looks like this:
```go
import as "github.com/aerospike/aerospike-client-go"

// ...

stm := as.NewStatement(namespace, set)
stm.Addfilter(as.NewEqualFilter("hasValue", 1))

policy := as.NewQueryPolicy()
policy.Priority = as.LOW // change policy here
recordSet, err := client.Query(policy, stm)
if err != nil {
	log.Panicf("failed to query: %s", err)
}
// read recordSet here
```

I tried different prorities `as.DEFAULT`, `as.MEDIUM` and `as.LOW`, hoping to see different behaviors. But I noticed that they all seem to behave exactly the same. If I look at the query job details in the aerospike dashboard I always see `priority: 10`. In my own query logs I see the same throughput (90k records per second) and the test traffic I'm producing drops from 100k to 80k reads/writes per second.

We're using a test cluster of 17 nodes running version 3.9.1. The test traffic and query go to a namespace with around 3B master objects while another (in this test unused) namespace contains another 5B master objects.

Looking at the Go client code the priority is never used internally, but just forwarded to the server. So if this is a server issue, please escalate it accordingly.

I would have expected to see some differences when using different priority values. What am I missing here? Thank you!

**Comment by khaf** [comment]
I'm still on this. Will get back to you as soon as I have an update.

**Comment by khaf** [comment]
Well it seems today, there's no controllable policy from the client perspective for queries.

There's server side control between short running queries and long running queries.

**Comment by wellle** [comment]
Alright, thank you! It looks like for our use case the query command works well. But out of curiosity, are you planning to make it possible to set different priorities for query commands in the future?

**Comment by khaf** [comment]
[Seems I had forgotten to press send on the answer to this question]

We are significantly improving the features of our scan/queries, but setting a `priority` for queries is not one of them.

**Comment by wellle** [comment]
@khaf: Thanks for the update. Here's a quick FYI:

After we added our indexes to our production nodes everything was working fine. But once we needed to restart some nodes (for updating etc) we noticed that instead of one hour, it now took around 10 hours to restart a single node. So updating our 20 nodes takes considerably longer now. Also while restarting, the replication factor is effectively reduced, so another node going down would already mean data loss with a replication factor of 2. So that didn't really feel safe for us and we decided to drop the indexes again and go for another solution.
